contrib_bots/lib/foursquare.py

```diff
@@ -0,0 +1,130 @@
+        with open(CONFIG_PATH) as settings:
+            config = cp.ConfigParser()
+            config.readfp(settings)
+            return config.get('FourSquare', 'api_key')
+
+    def format_json(self, text, location):
+        template = ("Food nearby {location} coming right up:")
+
+        venue_list = text['response']['venues']
+        venue_details = {}
+        for i in range(len(text['response']['venues'])):
+            template += '\n>{s}name_{i}\n>{s}formattedAddress_{i}\n>{s}descrip_{i}\n'.format(i=str(i) + '}', s='{')
+            place = venue_list[i]
+            venue_details['name_'+str(i)] = place['name']
+            venue_details['formattedAddress_'+str(i)] = ', '.join(place['location']['formattedAddress'])
+            venue_details['descrip_'+str(i)] = place['categories'][0]['pluralName']
+        venue_details['location'] = location
+        print(template)
+        return template.format(**venue_details)
+
+    def send_info(self, message, letter, client):
+        if message['type'] == 'private':
+            client.send_message(dict(
+                type='private',
+                to=message['sender_email'],
+                content=letter,
+            ))
+        else:
+            client.send_message(dict(
+                type='stream',
+                subject=message['subject'],
+                to=message['display_recipient'],
+                content=letter,
+                ))
+
+    def handle_message(self, message, client, state_handler):
+        words = message['content'].split()
+        if "/help" in words:
+            self.send_info(message, self.help_info, client)
+            return
+
+        try:
+            params = {'limit': '3'}
+            params['near']  = re.search('\'[A-Za-z]\w+[,]?[\s\w+]+?\'', message['content']).group(0)
+            params['radius'] = re.search('([0-9])\w+', message['content']).group(0)
+            params['query'] = re.search('\s(\w+)$', message['content']).group(0)
+            params['v'] = 20170108
+            params['oauth_token'] = self.api_key
+        except AttributeError:
+            params = {}
+
+        response = requests.get('https://api.foursquare.com/v2/venues/search?',
+                                params=params)
+        if response.status_code == 200:
+            received_json = response.json()
+        else:
+            self.send_info(message,
+                           "Invalid Request\nIf stuck, try '@foursquare help'.",
+                           client)
+            return
+
+        if received_json['meta']['code'] == 200:
+            response_msg = self.format_json(received_json, params['near'])
+            self.send_info(message, response_msg, client)
+            return
+        self.send_info(message,
+                       "Invalid Request\nIf stuck, try '@foursquare help'.",
+                       client)
+        return
+
+handler_class = FourSquareHandler
+handler = FourSquareHandler()
```

Comment:
If we pull `get_api_key()` up to the module level, we won't need to instantiate `handler` here for the tests.  And we should only do this in the tests, anyway.